kasa/discover.py

```diff
@@ -344,6 +344,7 @@ async def connect_single(
         port: Optional[int] = None,
         timeout=5,
         credentials: Optional[Credentials] = None,
+        protocol_id: int = 1,
```

Comment:
I'd rather to use `protocol_class=None` (or something similarly named) here, and initialize & set it if not none.

Reply:
Done